Junit5 migration

MoonBow-1 opened the pull request:
- [x] Rebase main to this branch, after #15 is merged

- Migrates JUnit 4 assertions to JUnit 5 assertions
- Will close #14

MoonBow-1 committed:
Remove unused variables and thrown exception in ConfigurationIT

Remove unnecessary throws from ServiceAndEventManagerIT.cleanup

Fix badly used variable reassignment in AckManagerIT.getCurrentAckValueTest

Remove all try-catches in AckManagerIT.java

Remove all try-catches in ServiceAndEventManagerIT.java

Use assertNotEquals in AckManagerIT.deleteAckTest instead of assertNotSame

Fix unused test

- Now in use

Use assertDoesNotThrow in ServiceAndEventManagerIT.initialize instead of try-catch

Remove duplicate property in AckManagerIT.java

Update Apache Rat pattern and exclusions

Run spotless:apply

Update the copyright year and formatting

Remove JUnit4 vintage-engine from dependencies

Migrate JUnit4 assertions to JUnit5 in TokenManagerTests

Migrate JUnit4 assertions to JUnit5 in SessionManagerTests

Migrate JUnit4 assertions to JUnit5 in ConverterTests

- Also includes minor enhancements to tests

Migrate JUnit4 assertions to JUnit5 in ServiceAndEventManagerIT

Fix tests in SendEventsIT

Migrate JUnit4 assertions to JUnit5 in AckManagerIT

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
 
         // Gets the SDParams from the SDEs from the first SyslogMessage returned from
         // Converter and saves them in a List
         for (iterator = returnedSDElements1.iterator(); iterator.hasNext();) {
```
Please add an assertion after each for-loop that an expected number of loops has been iterated

MoonBow-1 replied:
New issue #32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
 
         // Gets the SDParams from the SDEs from the first SyslogMessage created in
         // initialize() and saves them in a List
         for (iterator = supposedSDElements1.iterator(); iterator.hasNext();) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
-            assertTrue(
-                    "SDParam '" + returnedSDParams.get(i) + "' should not be in returned SDElement.", supposedSDParams.contains(returnedSDParams.get(i))
-            );
+        for (final SDParam returnedSDParam : returnedSDParams) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
-            assertTrue(
-                    "SDParam '" + supposedSDParams.get(i) + "' should be in returned SDElement.", returnedSDParams.contains(supposedSDParams.get(i))
-            );
+        for (final SDParam supposedSDParam : supposedSDParams) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
 
         // Gets the SDParams from the SDEs from the second SyslogMessage returned from
         // Converter and saves them in a List
         for (iterator = returnedSDElements2.iterator(); iterator.hasNext();) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
 
         // Gets the SDParams from the SDEs from the second SyslogMessage created in
         // initialize() and saves them in a List
         for (iterator = supposedSDElements2.iterator(); iterator.hasNext();) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
-            assertTrue(
-                    "SDParam '" + returnedSDParams.get(i) + "' should not be in returned SDElement.", supposedSDParams.contains(returnedSDParams.get(i))
-            );
+        for (SDParam returnedSDParam : returnedSDParams) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
-            assertTrue(
-                    "SDParam '" + supposedSDParams.get(i) + "' should be in returned SDElement.", returnedSDParams.contains(supposedSDParams.get(i))
-            );
+        for (final SDParam supposedSDParam : supposedSDParams) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
 
         // Gets the SDParams from the SDEs from the third SyslogMessage returned from
         // Converter and saves them in a List
         for (iterator = returnedSDElements3.iterator(); iterator.hasNext();) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
 
         // Gets the SDParams from the SDEs from the third SyslogMessage created in
         // initialize() and saves them in a List
         for (iterator = supposedSDElements3.iterator(); iterator.hasNext();) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
-            assertTrue(
-                    "SDParam '" + returnedSDParams.get(i) + "' should not be in returned SDElement.", supposedSDParams.contains(returnedSDParams.get(i))
-            );
+        for (final SDParam returnedSDParam : returnedSDParams) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/ConverterTests.java:
```diff
@@ -295,118 +322,102 @@ public void timestampTest() {
-            assertTrue(
-                    "SDParam '" + supposedSDParams.get(i) + "' should be in returned SDElement.", returnedSDParams.contains(supposedSDParams.get(i))
-            );
+        for (final SDParam supposedSDParam : supposedSDParams) {
```
assert expected loops here also

MoonBow-1 replied:
Issue for this: https://github.com/teragrep/cfe_16/issues/32

elliVM commented on src/test/java/com/teragrep/cfe_16/it/AckManagerIT.java:
```diff
@@ -82,6 +79,7 @@
         "max.ack.age=20000",
         "max.session.age=30000",
         "poll.time=30000",
+        "poll.time=30000",
```
is this duplicate prop `poll.time=30000` added on purpose or should it be different?

MoonBow-1 replied:
Fixed in commit 5c36e4a8

elliVM commented on src/test/java/com/teragrep/cfe_16/it/SendEventsIT.java:
```diff
@@ -79,18 +88,16 @@
 @SpringBootTest
 public class SendEventsIT implements Runnable {
 
+    private final static int SERVER_PORT = 1236;
```
Please use conventional ordering `private static final`

MoonBow-1 replied:
Will be fixed in #15, commit 42cb8a156d30da6f48075b4d2c56fe1b6483d5b9

elliVM commented on src/test/java/com/teragrep/cfe_16/it/ServiceAndEventManagerIT.java:
```diff
@@ -88,46 +94,26 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ServiceAndEventManagerIT.class);
-    private static Server server;
     private static final String hostname = "localhost";
```
`hostname` is unused

MoonBow-1 replied:
Will be fixed by #15

elliVM commented on src/test/java/com/teragrep/cfe_16/it/ServiceAndEventManagerIT.java:
```diff
@@ -137,15 +123,30 @@ public static void cleanup() throws InterruptedException {
+    }
+
+    @AfterAll
+    public static void cleanup() throws InterruptedException {
```
remove throws

MoonBow-1 replied:
Fixed in commit 1feb1142

elliVM commented on src/test/java/com/teragrep/cfe_16/it/ServiceAndEventManagerIT.java:
```diff
@@ -137,15 +123,30 @@ public static void cleanup() throws InterruptedException {
+    }
 
     private static ServerSocket getSocket() {
         ServerSocket socket = null;
```
null assignment not required

MoonBow-1 replied:
Fixed in commit 96637031

elliVM commented on src/test/java/com/teragrep/cfe_16/it/ServiceAndEventManagerIT.java:
```diff
@@ -400,22 +401,29 @@ public void handleTimeTest() {
-        String content8 = "{\"event\": \"Pony 1 has left the barn\", \"sourcetype\": \"mysourcetype\", \"time\": 1433188255252321}";
+        String content8 = "{\"event\": \"Pony 1 has left the barn\", \"sourcetype\": "
+                + "\"mysourcetype\", \"time\": 1433188255252321}";
 
```
null assignments of nodes are not required here

MoonBow-1 replied:
Fixed in commit 96637031

elliVM commented on src/test/java/com/teragrep/cfe_16/it/ServiceAndEventManagerIT.java:
```diff
@@ -568,11 +626,17 @@ public void sendingMultipleEventsTest() {
      */
     public void sendingMultipleEventsWithDefaultChannelTest() {
```
This seems to be an unused method

MoonBow-1 replied:
Will be fixed by https://github.com/teragrep/cfe_16/pull/15/

elliVM commented on src/test/java/com/teragrep/cfe_16/it/SendEventsIT.java:
```diff
@@ -79,18 +88,16 @@
-    private final static int SERVER_PORT = 1236;
     private CountDownLatch countDownLatch = new CountDownLatch(0);
 
     @BeforeEach
```
Could we use `Assert.assertDoesNotThrow` instead of throws IOException here?

MoonBow-1 replied:
Will be fixed by https://github.com/teragrep/cfe_16/pull/15

elliVM commented on src/test/java/com/teragrep/cfe_16/it/SendEventsIT.java:
```diff
@@ -79,18 +88,16 @@
     private MockHttpServletRequest request1;
     private String eventInJson;
     private String channel1;
     private Selector selector;
```
`selector` is never accessed

MoonBow-1 replied:
Will be fixed by https://github.com/teragrep/cfe_16/pull/15

elliVM commented on src/test/java/com/teragrep/cfe_16/it/SendEventsIT.java:
```diff
@@ -140,7 +150,7 @@ public void run() {
 
     @Test
-    public void sendEventsTest() throws IOException, InterruptedException, ExecutionException {
+    public void sendEventsTest() throws InterruptedException, ExecutionException {
```
Same here is it possible to not throw exceptions?

MoonBow-1 replied:
Will be fixed by https://github.com/teragrep/cfe_16/pull/15

elliVM commented on src/test/java/com/teragrep/cfe_16/it/SendEventsIT.java:
```diff
@@ -174,16 +184,17 @@ public void sendEventsTest() throws IOException, InterruptedException, Execution
     @Disabled
     @Test
-    public void send1EventTest() throws IOException, InterruptedException {
+    public void send1EventTest() throws InterruptedException {
```
Same here is it possible to not throw exceptions?

MoonBow-1 replied:
Will be fixed by https://github.com/teragrep/cfe_16/pull/15

elliVM commented on src/test/java/com/teragrep/cfe_16/it/AckManagerIT.java:
```diff
@@ -194,8 +184,8 @@ public void getRequestedAckStatusesTest() {
         JsonNode faultyNode = mapper.createObjectNode();
         JsonNode notIntNode = mapper.createObjectNode();
```
`queryNode`, `faultyNode` and `notIntNode` are reassigned later so assignment is redundant here

MoonBow-1 replied:
Was fixed in commit afc0909f

elliVM commented on src/test/java/com/teragrep/cfe_16/it/AckManagerIT.java:
```diff
@@ -194,8 +184,8 @@ public void getRequestedAckStatusesTest() {
-        assertTrue(ackManager.acknowledge(this.authToken1, this.channel1, 1));
+        Assertions.assertTrue(ackManager.addAck(this.authToken1, this.channel1, new Ack(1, false)));
+        Assertions.assertTrue(ackManager.acknowledge(this.authToken1, this.channel1, 1));
         try {
```
use `Assertions.assertDoesNotThrow` instead of try

MoonBow-1 replied:
Fixed in commit afc0909f

elliVM commented on src/test/java/com/teragrep/cfe_16/it/AckManagerIT.java:
```diff
@@ -204,78 +194,77 @@ public void getRequestedAckStatusesTest() {
+        Assertions
+                .assertEquals(emptyJsonNode, ackManager.getRequestedAckStatuses(this.authToken1, "", null), "getRequestedAckStatuses should return null, when providing a null value as a " + "parameter");
 
         try {
```
use Assertions.assertDoesNotThrow instead of try

MoonBow-1 replied:
Fixed in commit b422c3a7

MoonBow-1 replied:
Fixed in commit https://github.com/teragrep/cfe_16/commit/afc0909f15e72124d8690cd710c4fc0718404021

elliVM commented on src/test/java/com/teragrep/cfe_16/it/AckManagerIT.java:
```diff
@@ -204,78 +194,77 @@ public void getRequestedAckStatusesTest() {
+                );
     }
 
     public void getCurrentAckValueAndIncrementTest() {
```
unused method, could this be removed?

MoonBow-1 replied:
Taken into use in commit 909965a8

elliVM commented on src/test/java/com/teragrep/cfe_16/it/AckManagerIT.java:
```diff
@@ -292,28 +281,28 @@ public void deleteAckTest() {
-        assertNotSame("Ack lists should not be same", list1.toString(), list2.toString());
-        assertEquals("list2 should be shorter by one index", list1Size - 1, list2Size);
-        assertFalse("list2 should not contain the deleted ack", list2.containsKey(deletedAck.getId()));
+        Assertions.assertNotSame(list1.toString(), list2.toString(), "Ack lists should not be same");
```
Could this be `assertNotEquals` and without `toString()` call to compare the map values?

MoonBow-1 replied:
Fixed in commit f9b98e43

elliVM commented on src/test/java/com/teragrep/cfe_16/it/SendEventsIT.java:
```diff
@@ -160,7 +170,7 @@ public void sendEventsTest() throws IOException, InterruptedException, Execution
         for (Future<String> f : futures) {
```
could add assertion for expected number of `countFuture`

MoonBow-1 replied:
Will be fixed in #15 or in #32

elliVM reviewed (changes requested):
Left some comments about the general quality of the unit tests which might be outside of to the scope of this PR. So you can resolve them with an new issue.

elliVM reviewed (dismissed):
looks good to me

eemhu reviewed (dismissed):
junit5 migration wise looks OK.

MoonBow-1 committed:
Delete empty file after rebase

MoonBow-1 committed:
Remove old JUnit 4 assertions

- Removes the static imports
- Changes the order of the assertion parameters to the new format with message being last

MoonBow-1 committed:
Fix the changed copyright in SyslogMessageSender

- Noticed that the rebase container modifications to the copyright, so reverted those

MoonBow-1 committed:
Move import to correct position in SessionManagerTests after rebase
